.github/scripts/get-job-id.ts

```diff
@@ -0,0 +1,44 @@
+import { retry } from './shared/utils';
+import { z } from 'zod';
+import * as core from '@actions/core';
+
+async function main() {
+  const { Octokit } = await import('octokit');
+
+  const env = z
+    .object({
+      OWNER: z.string().default('metamask'),
+      REPOSITORY: z.string().default('metamask-extension'),
+      RUN_ID: z.coerce.number(),
+      JOB_NAME: z.string(),
+      ATTEMPT_NUMBER: z.coerce.number(),
+      GITHUB_TOKEN: z.string(),
+      GITHUB_ACTIONS: z.coerce.boolean().default(false),
+    })
+    .parse(process.env);
```

Comment:
Why not do something like:

```
const env = {
  OWNER: process.env.OWNER || 'metamask',
  REPOSITORY: process.env.REPOSITORY || 'metamask-extension',
  RUN_ID: Number(process.env.RUN_ID),
  JOB_NAME: process.env.JOB_NAME,
  ATTEMPT_NUMBER: Number(process.env.ATTEMPT_NUMBER),
  GITHUB_TOKEN: process.env.GITHUB_TOKEN,
  GITHUB_ACTIONS: process.env.GITHUB_ACTIONS === 'true',
};
```

Do we actually need the type checking zod is doing?

Reply:
>wasting precious compute time on CI

it'll take like < 1ms more to find out.

> so having bulletproof validation makes CI safer

Increase our reliance on external dependencies makes CI _less safe_, since it increases the risk of supply chain attacks.

Reply:
@NicholasEllul Do you have any thoughts or concerns from a security standpoint on us using external npm packages in CI workflows like we are with `zod` here?

Reply:
just FYI: `yarn install` shouldn't be running any scripts unless we've explicitly allowed them via our `lavamoat.allowScripts` package.json config.